src/deploy/functions/prepareFunctionsUpload.ts

```diff
@@ -88,7 +90,8 @@ async function packageSource(
       });
     }
     if (typeof runtimeConfig !== "undefined") {
-      const runtimeConfigString = JSON.stringify(runtimeConfig, null, 2);
```

Comment:
There is a chance mutating the runtimeConfig string may break function upload behavior.
(@taeold / @inlined can keep me honest here.)

To mitigate that, can you use a separate variable to push into the hash?

```
const runtimeConfigString = JSON.stringify(runtimeConfig, null, 2);
      // In order for hash to be consistent, configuration object tree must be sorted by key, only possible with arrays.
 const runtimeConfigHashString = JSON.stringify(convertToSortedKeyValueArray(runtimeConfig));
hashes.push(runtimeConfigHashString);
```

Reply:
IIUC, this code is mutating the value that we pass into the hash, not the underlying runtime config value.

If I understood correctly, that looks safe!

Reply:
@TheIronDev @taeold I think it makes sense for the configuration file in the archive to be pretty printed anyways. I went ahead and added a separate formatted string variable. 

Also, as suggested, added a few tests for the sort function. Please have a look @TheIronDev.